src/algorithm/center-of-mass-derivatives.hxx

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2019 CNRS
+//
+
+#ifndef __pinocchio_center_of_mass_derivatives_hxx__
+#define __pinocchio_center_of_mass_derivatives_hxx__
+
+#include "pinocchio/multibody/visitor.hpp"
+#include "pinocchio/algorithm/check.hpp"
+
+namespace pinocchio
+{
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl,
+           typename Matrix3xOut>
+  struct CoMVelocityDerivativesForwardStep
+    : public fusion::JointVisitorBase< CoMVelocityDerivativesForwardStep<Scalar,Options,JointCollectionTpl,Matrix3xOut> >
+  {
+    /* Assumes that both computeForwardKinematicsDerivatives and centerOfMass (or
+     * equivalent methods like computeAllTerms) have been computed beforehand. */
+
+    typedef ModelTpl<Scalar,Options,JointCollectionTpl> Model;
+    typedef DataTpl<Scalar,Options,JointCollectionTpl> Data;
+    
+    typedef boost::fusion::vector<const Model &,
+                                  Data &,
+                                  Matrix3xOut &
+                                  > ArgsType;
+    
+    template<typename JointModel>
+    static void algo(const JointModelBase<JointModel> & jmodel,
+                     JointDataBase<typename JointModel::JointDataDerived> & jdata,
+                     const Model & model,
+                     Data & data,
+                     const Eigen::MatrixBase<Matrix3xOut> & vcom_partial_dq_CONST)
```

Comment:
`vcom_partial_dq_CONST` is not really elegant. I would keep `vcom_partial_dq`

Reply:
I don't agree. In other algo, you you XXX for the const ref and XXX_ for the const-cast ref. I think this convention is better. 
Never mind. I changed it.